### PR TITLE
Router transform added to all destinations

### PR DESCRIPTION
**Pull request by shrouti1507**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by arnab-p — APPROVED** [review]

File: jest.config.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ module.exports = {
   coverageDirectory: "coverage",
 
   // An array of regexp pattern strings used to skip coverage collection
-  coveragePathIgnorePatterns: ["/node_modules/"],
+  coveragePathIgnorePatterns: ["/node_modules/", "__tests__", "warehouse/v0"],
```
Comment (ruchiramoitra) [inline jest.config.js:30]: why are we ignoring this?

**Review by ruchiramoitra — COMMENTED** [review]
segment as destination should also be added I think. Are we not doing for warehouse destinations?
Whenever there is no error thrown i see router transform is not impleted. Is that what we are doing? 
If many payloads are returned by transformer how will that be handled?
@arnab-p 

Other changes lgtm

**Review by ruchiramoitra — COMMENTED** [review]
Why are we not doing app centre, af, auth 0,customer io?

File: v0/destinations/ga/transform.js [hunk 2]
```diff
@@ -876,7 +880,7 @@ function process(event) {
   try {
     response = processSingleMessage(event.message, event.destination);
   } catch (error) {
-    throw new Error(error.message || "Unknown error");
+    throw new Error(error.message || "Unknown error", error.status || 400);
```
Comment (ruchiramoitra) [inline v0/destinations/ga/transform.js:883]: this should be custom

**Review by ruchiramoitra — COMMENTED** [review]
fb destination line number 158 check.
eventbridge line number  44.

**Review by ruchiramoitra — COMMENTED** [review]
Should we also change the errors thrown in common utils file? @arnab-p

**Comment by arnab-p** [comment]
> Should we also change the errors thrown in common utils file? @arnab-p

Yes that's a good idea

**Review by arnab-p — APPROVED** [review]

**Pull request merged**